app/src/main/java/com/example/sitstandtimer/data/UserPreferenceRepository.kt

```diff
@@ -17,132 +17,92 @@ class UserPreferenceRepository(
     private val dataStore: DataStore<Preferences>
 ) {
     private companion object {
-        val NUMBER_OF_INTERVALS = intPreferencesKey("number_of_intervals")
-        val INTERVAL_LENGTH = floatPreferencesKey("interval_length")
-        val BREAK_LENGTH = floatPreferencesKey("break_length")
-        val LUNCH_LENGTH = floatPreferencesKey("lunch_length")
-        val SNOOZE_LENGTH = floatPreferencesKey("snooze_length")
-        val REMOTE_NFC_TAG = stringPreferencesKey("remote_nfc_tag")
-        val DESK_NFC_TAG = stringPreferencesKey("desk_nfc_tag")
         const val TAG = "UserPreferencesRepo"
-    }
 
-    suspend fun savePreferences(
-        numberOfIntervals: Int,
-        intervalLength: Float,
-        breakLength: Float,
-        lunchLength: Float,
-        snoozeLength: Float
-    ) {
-        dataStore.edit { preferences ->
-            preferences[NUMBER_OF_INTERVALS] = numberOfIntervals
-            preferences[INTERVAL_LENGTH] = intervalLength
-            preferences[BREAK_LENGTH] = breakLength
-            preferences[LUNCH_LENGTH] = lunchLength
-            preferences[SNOOZE_LENGTH] = snoozeLength
-        }
+        // Keys
+        private val NUMBER_OF_INTERVALS = intPreferencesKey("number_of_intervals")
+        private val INTERVAL_LENGTH = floatPreferencesKey("interval_length")
+        private val BREAK_LENGTH = floatPreferencesKey("break_length")
+        private val LUNCH_LENGTH = floatPreferencesKey("lunch_length")
+        private val SNOOZE_LENGTH = floatPreferencesKey("snooze_length")
+        private val REMOTE_NFC_TAG = stringPreferencesKey("remote_nfc_tag")
+        private val DESK_NFC_TAG = stringPreferencesKey("desk_nfc_tag")
+
+        // Default Values
+        private const val DEFAULT_NUMBER_OF_INTERVALS = 2
+        private const val DEFAULT_INTERVAL_LENGTH = 30f
+        private const val DEFAULT_BREAK_LENGTH = 15f
+        private const val DEFAULT_LUNCH_LENGTH = 60f
+        private const val DEFAULT_SNOOZE_LENGTH = 15f
+        private const val DEFAULT_NFC_TAG = ""
     }
 
-    val numberOfIntervals: Flow<Int> = dataStore.data
+    private fun <T> Flow<Preferences>.getPreferenceWithFallback(
+        key: Preferences.Key<T>,
+        defaultValue: T
+    ): Flow<T> = this
         .catch {
-            if(it is IOException) {
+            if (it is IOException) {
                 Log.e(TAG, "Error reading preferences.", it)
                 emit(emptyPreferences())
             } else {
                 throw it
             }
         }
         .map { preferences ->
-            preferences[NUMBER_OF_INTERVALS] ?: 2
+            preferences[key] ?: defaultValue
         }
```

Comment:
This function uses generics to allow for more readable `dataStore.data.getPreferenceWithFallback(...)`